In [1]:
# YOLOv11 Training Script with Full Parameter Explanations and Improved Stability

from ultralytics import YOLO
import os
from datetime import datetime

# --- Model Setup ---
# Load a pretrained YOLOv8n model (nano version: fastest, smallest, good for limited data or edge devices)
model = YOLO('../src/human_detector/models/yolov11n.pt')
# To resume training from a previous checkpoint:
# model = YOLO('/path/to/your/best.pt')

# --- Training Configuration ---
epochs = 300               # Number of times the model sees the entire dataset
imgsz = 640                # Size to which all images are resized (square). 640 is a good trade-off between speed and accuracy
batch_size = 4             # Number of images used in one forward/backward pass (small for limited VRAM)

lr0 = 1e-5                 # Initial learning rate (controls how much the model updates each step)
lrf = 0.01                 # Final learning rate = lr0 * lrf, used with cosine decay learning schedule
optimizer = "Adam"          # Optimizer algorithm: 'SGD' is more stable for small datasets, 'AdamW' (default) adapts better to large, noisy datasets
# warmup_epochs = 3          # Number of epochs to slowly ramp up learning rate to avoid instability at start
dropout = 0.1              # Dropout rate (randomly turns off neurons during training to prevent overfitting)

# --- Experiment Folder Naming ---
time_stamp = datetime.now().strftime("%Y%m%d_%H%M")

exp_name = (
    f"yolov11n_v2_rect_false_"         # Model name and version. You can use this to identify custom model variants.
    f"ep{epochs}_"          # Number of epochs (e.g., ep300 means trained for 300 full dataset passes)
    f"img{imgsz}_"          # Image input size (e.g., img640 means all images resized to 640x640)
    f"bs{batch_size}_"      # Batch size (e.g., bs4 means 4 images processed per training step)
    f"lr{lr0}_"             # Initial learning rate (e.g., lr0.0001)
    f"lrf{lrf}_"            # Final learning rate factor (e.g., lrf0.01 → lr drops to 0.000001 at end)
    f"{optimizer}_"         # Optimizer used (e.g., SGD, AdamW, etc.)
    f"augmented_"            # Tag for augmentation setup (e.g., Final or custom profile name)
    f"{time_stamp}"         # Timestamp for uniqueness and chronological sorting
)

# --- Training Start ---
results = model.train(
    seed=42,                     # Random seed for reproducibility
    data='data.yaml',            # Path to data config file (defines class names and image paths)
    epochs=epochs,               # Total number of training epochs
    patience=50,                 # Early stopping: if val loss doesn't improve for 10 epochs, stop training
    dropout=dropout,             # Helps prevent overfitting by randomly disabling neurons
    imgsz=imgsz,                 # Resizes all images to this square dimension
    batch=batch_size,            # Number of images per training batch
    lr0=lr0,                     # Starting learning rate
    lrf=lrf,                     # Final learning rate multiplier
    # warmup_epochs=warmup_epochs,# Smooth ramp-up of LR to avoid unstable gradients
    optimizer=optimizer,         # Optimizer choice: SGD recommended for small datasets
    device=0,                    # GPU device ID (0 = first GPU)
    half=True,                   # Use float16 precision if supported (faster training)
    workers=2,                   # Number of CPU workers for data loading (adjust based on CPU)
    cache=True,                  # Preload and cache images in memory for faster epoch execution
    # rect=True,                   # Enables rectangular training for better image aspect ratio handling
    verbose=True,                # Print detailed training logs for every batch
    amp=False ,
    # exist_ok=True,               # Overwrites existing output folder if it already exists

    # --- Data Augmentation (tuned for small datasets) ---
    hsv_h=0.015,                 # Hue variation (color tint)
    hsv_s=0.7,                   # Saturation variation (color strength)
    hsv_v=0.4,                   # Brightness variation
    translate=0.05,              # Random image shift (reduced for safety)
    scale=0.1,                   # Random zoom in/out (less aggressive)
    fliplr=0.5,                  # 50% chance to flip images horizontally
    mosaic=0.2,                  # Combines 4 training images into one
    mixup=0.03,                  # Blends two images and labels
    copy_paste=0.03,             # Copies object from one image and pastes onto another
    perspective=0.001,           # Small perspective distortion to improve robustness

    # --- Output Settings ---
    project='results',           # Base directory to store training logs and weights
    name=exp_name                # Subfolder name under project/ to store this run’s output
)

# --- Save Path Display ---
save_dir = str(results.save_dir)
best_pt = os.path.join(save_dir, 'weights', 'best.pt')
print(f"✅ Trained checkpoint saved to: {best_pt}")


New https://pypi.org/project/ultralytics/8.3.146 available 😃 Update with 'pip install -U ultralytics'
Ultralytics 8.3.141 🚀 Python-3.10.12 torch-2.7.0+cu126 CUDA:0 (NVIDIA GeForce MX550, 1864MiB)
engine/trainer: agnostic_nms=False, amp=False, augment=False, auto_augment=randaugment, batch=4, bgr=0.0, box=7.5, cache=True, cfg=None, classes=None, close_mosaic=10, cls=0.5, conf=None, copy_paste=0.03, copy_paste_mode=flip, cos_lr=False, cutmix=0.0, data=data.yaml, degrees=0.0, deterministic=True, device=0, dfl=1.5, dnn=False, dropout=0.1, dynamic=False, embed=None, epochs=300, erasing=0.4, exist_ok=False, fliplr=0.5, flipud=0.0, format=torchscript, fraction=1.0, freeze=None, half=True, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, imgsz=640, int8=False, iou=0.7, keras=False, kobj=1.0, line_width=None, lr0=1e-05, lrf=0.01, mask_ratio=4, max_det=300, mixup=0.03, mode=train, model=../src/human_detector/models/yolov11n.pt, momentum=0.937, mosaic=0.2, multi_scale=False, name=yolov11n_v2_rect_false_ep300_i

train: Scanning /home/yogee/Desktop/human_detector_ws/data_training/data_person/train/labels.cache... 4044 images, 23 backgrounds, 0 corrupt: 100%|██████████| 4044/4044 [00:00<?, ?it/s]


WARNING ⚠️ cache='ram' may produce non-deterministic training results. Consider cache='disk' as a deterministic alternative if your disk space allows.


train: Caching images (3.3GB RAM): 100%|██████████| 4044/4044 [00:12<00:00, 324.47it/s] 


val: Fast image access ✅ (ping: 0.0±0.0 ms, read: 366.2±215.5 MB/s, size: 448.1 KB)


val: Scanning /home/yogee/Desktop/human_detector_ws/data_training/data_person/test/labels.cache... 1720 images, 44 backgrounds, 0 corrupt: 100%|██████████| 1720/1720 [00:00<?, ?it/s]


WARNING ⚠️ cache='ram' may produce non-deterministic training results. Consider cache='disk' as a deterministic alternative if your disk space allows.


val: Caching images (1.4GB RAM): 100%|██████████| 1720/1720 [00:04<00:00, 408.29it/s]
/home/yogee/.local/lib/python3.10/site-packages/matplotlib/projections/__init__.py:63: UserWarning: Unable to import Axes3D. This may be due to multiple versions of Matplotlib being installed (e.g. as a system package and as a pip package). As a result, the 3D projection is not available.
  warnings.warn("Unable to import Axes3D. This may be due to multiple versions of "


Plotting labels to results/yolov11n_v2_rect_false_ep300_img640_bs4_lr1e-05_lrf0.01_Adam_augmented_20250529_2230/labels.jpg... 
optimizer: Adam(lr=1e-05, momentum=0.937) with parameter groups 81 weight(decay=0.0), 88 weight(decay=0.0005), 87 bias(decay=0.0)
Image sizes 640 train, 640 val
Using 2 dataloader workers
Logging results to results/yolov11n_v2_rect_false_ep300_img640_bs4_lr1e-05_lrf0.01_Adam_augmented_20250529_2230
Starting training for 300 epochs...

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      1/300       1.1G      1.274      1.264      1.354         26        640: 100%|██████████| 1011/1011 [03:49<00:00,  4.40it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00,  9.83it/s]


                   all       1720       3999      0.869      0.815      0.861      0.574

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      2/300      1.15G      1.149     0.9995      1.266         12        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.06it/s]


                   all       1720       3999      0.889      0.851      0.892      0.619

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      3/300      1.15G      1.081     0.9124      1.216         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.11it/s]


                   all       1720       3999      0.904      0.858        0.9      0.636

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      4/300      1.14G      1.051     0.8689      1.207          8        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]


                   all       1720       3999      0.904      0.865      0.903      0.643

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      5/300      1.14G      1.034     0.8472      1.189         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.32it/s]


                   all       1720       3999      0.919       0.86      0.906      0.648

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      6/300      1.14G      1.015     0.8224      1.181         14        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.13it/s]


                   all       1720       3999      0.918      0.861      0.909       0.66

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      7/300      1.14G     0.9938     0.8001      1.165          9        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]


                   all       1720       3999      0.917      0.865      0.908      0.664

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      8/300      1.14G     0.9847     0.7942      1.158         35        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]


                   all       1720       3999       0.92      0.865       0.91       0.67

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


      9/300      1.14G     0.9774     0.7871      1.153          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]


                   all       1720       3999      0.916      0.868      0.908      0.676

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     10/300      1.14G     0.9563     0.7639      1.147         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.18it/s]


                   all       1720       3999      0.921       0.87      0.914      0.682

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     11/300      1.14G      0.955     0.7623      1.145         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.11it/s]


                   all       1720       3999      0.925      0.869      0.913      0.682

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     12/300      1.15G     0.9423     0.7471      1.142          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]


                   all       1720       3999      0.921      0.872      0.913      0.688

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     13/300      1.14G     0.9347     0.7334      1.131         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.30it/s]

                   all       1720       3999      0.929      0.869      0.915      0.694



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     14/300      1.14G     0.9266     0.7271      1.126         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.27it/s]


                   all       1720       3999      0.927       0.87      0.915      0.692

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     15/300      1.14G     0.9171     0.7244      1.119          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.13it/s]


                   all       1720       3999      0.922      0.876      0.916      0.691

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     16/300      1.14G     0.9121     0.7198      1.121         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]


                   all       1720       3999      0.926      0.872      0.914      0.697

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     17/300      1.14G     0.9043     0.7131      1.115          8        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.23it/s]


                   all       1720       3999      0.921      0.879      0.918      0.697

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     18/300      1.14G     0.8971     0.7058      1.112         13        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.13it/s]


                   all       1720       3999      0.919      0.879      0.917      0.705

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     19/300      1.14G     0.8965     0.7064      1.113          9        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.07it/s]


                   all       1720       3999      0.922      0.883      0.918      0.699

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     20/300      1.14G     0.8857     0.6922      1.101          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]


                   all       1720       3999      0.924      0.882      0.919      0.713

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     21/300      1.14G     0.8945     0.6981      1.109         11        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]


                   all       1720       3999      0.926      0.881      0.918      0.711

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     22/300      1.14G     0.8719     0.6797      1.095         11        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.26it/s]

                   all       1720       3999       0.93      0.878      0.919      0.714



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     23/300      1.14G     0.8759     0.6829      1.101         18        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.25it/s]

                   all       1720       3999      0.924      0.884      0.921      0.719



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     24/300      1.14G     0.8707     0.6737      1.096          8        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]


                   all       1720       3999      0.922      0.883      0.919      0.715

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     25/300      1.14G      0.857     0.6647      1.088          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]


                   all       1720       3999      0.926      0.882      0.922      0.716

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     26/300      1.14G     0.8699     0.6693       1.09         16        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.26it/s]

                   all       1720       3999      0.921      0.885      0.921      0.717



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     27/300      1.14G     0.8516     0.6554      1.086          6        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.924      0.887      0.921       0.72



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     28/300      1.14G     0.8526     0.6557      1.089         14        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.915      0.891      0.921      0.725



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     29/300      1.14G     0.8597     0.6646       1.09         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]


                   all       1720       3999      0.927      0.882      0.922      0.726

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     30/300      1.14G     0.8532     0.6584      1.088         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.925      0.887      0.923      0.728



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     31/300      1.14G     0.8563     0.6583      1.087         18        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]

                   all       1720       3999      0.927      0.885      0.925      0.726



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     32/300      1.14G     0.8382     0.6458      1.078         13        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.926      0.887      0.924      0.731



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     33/300      1.14G      0.848      0.645       1.08         12        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.925      0.886      0.924      0.736



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     34/300      1.14G     0.8354     0.6333      1.078         16        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.18it/s]

                   all       1720       3999      0.933      0.882      0.926      0.733



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     35/300      1.14G      0.832     0.6318      1.075          9        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.933      0.882      0.926      0.736



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     36/300      1.14G     0.8364     0.6395      1.073         11        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.28it/s]

                   all       1720       3999       0.93      0.886      0.926      0.732



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     37/300      1.14G     0.8281     0.6296      1.071          7        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.27it/s]

                   all       1720       3999      0.924      0.891      0.926      0.734



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     38/300      1.14G     0.8201     0.6252       1.07         14        640: 100%|██████████| 1011/1011 [03:52<00:00,  4.35it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.29it/s]

                   all       1720       3999      0.925      0.889      0.926      0.736



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     39/300      1.14G     0.8221     0.6207      1.068          6        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.18it/s]

                   all       1720       3999      0.926      0.893      0.928      0.739



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     40/300      1.14G     0.8153     0.6146      1.069         15        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.924      0.889      0.926      0.738



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     41/300      1.14G      0.807     0.6085      1.059          7        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.924      0.893      0.926      0.738



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     42/300      1.14G     0.8158     0.6174      1.068         18        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.931      0.885      0.925      0.737



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     43/300      1.14G     0.8174     0.6133      1.062         11        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.929      0.886      0.923       0.74



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     44/300      1.14G     0.8093     0.6072      1.056         13        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.925      0.891      0.928      0.741



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     45/300      1.14G     0.8006     0.6095      1.058         10        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.924      0.893      0.929      0.747



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     46/300      1.14G     0.8116     0.6096      1.062          8        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.923      0.895      0.928      0.743



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     47/300      1.14G     0.7973     0.5943      1.058          9        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.24it/s]

                   all       1720       3999       0.92      0.897      0.928      0.745



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     48/300      1.14G     0.8027     0.6049      1.054         12        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.04it/s]

                   all       1720       3999      0.926      0.892      0.929      0.743



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     49/300      1.14G      0.794      0.597      1.056         14        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.09it/s]

                   all       1720       3999      0.924      0.895      0.929      0.746



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     50/300      1.14G     0.7885     0.5954      1.052          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.932      0.888      0.928      0.745



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     51/300      1.14G     0.7976     0.6014      1.052         11        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.929      0.888      0.928      0.749



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     52/300      1.14G      0.791     0.5928      1.048          6        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.14it/s]

                   all       1720       3999      0.925      0.893      0.929      0.751



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     53/300      1.14G     0.7934     0.5965      1.051         18        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.28it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.929       0.89       0.93      0.751



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     54/300      1.14G     0.7887     0.5898       1.05         29        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.926      0.893       0.93      0.752



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     55/300      1.14G     0.7833     0.5847      1.044          7        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.925      0.893      0.929      0.749



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     56/300      1.14G     0.7737     0.5795      1.042         13        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.929      0.891       0.93      0.754



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     57/300      1.14G     0.7747     0.5762      1.044         15        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.13it/s]

                   all       1720       3999      0.929      0.889       0.93       0.75



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     58/300      1.14G     0.7806     0.5791      1.044         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.924      0.895       0.93      0.754



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     59/300      1.14G     0.7763      0.579      1.042         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.921      0.896       0.93      0.751



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     60/300      1.14G     0.7784      0.576      1.041         10        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.928       0.89      0.928      0.753



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     61/300      1.14G     0.7684     0.5631      1.039         15        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.924      0.894      0.929      0.751



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     62/300      1.14G     0.7713     0.5757       1.04          8        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.925      0.892       0.93      0.751



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     63/300      1.14G     0.7582     0.5632      1.029         10        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.928      0.891       0.93      0.757



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     64/300      1.14G     0.7614     0.5736      1.035         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.934      0.885       0.93      0.756



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     65/300      1.14G     0.7691     0.5674      1.036         12        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.25it/s]

                   all       1720       3999      0.926       0.89      0.929      0.753



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     66/300      1.14G     0.7605     0.5656      1.036         12        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.922      0.892       0.93      0.756



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     67/300      1.14G     0.7568     0.5599      1.028          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.927      0.889      0.931      0.757



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     68/300      1.14G     0.7644     0.5634      1.036         23        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.927      0.889       0.93      0.756



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     69/300      1.14G     0.7561     0.5599      1.034         15        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.14it/s]

                   all       1720       3999      0.923      0.895      0.929      0.757



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     70/300      1.14G     0.7695     0.5658      1.039         11        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.932       0.89      0.931      0.759



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     71/300      1.14G     0.7496     0.5539      1.028         10        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.18it/s]

                   all       1720       3999      0.925      0.893      0.931      0.759



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     72/300      1.14G     0.7549     0.5573       1.03         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.932      0.885      0.931      0.758



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     73/300      1.14G     0.7542     0.5523      1.029         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.22it/s]

                   all       1720       3999       0.93      0.891      0.931       0.76



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     74/300      1.14G     0.7534     0.5532       1.03          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.924      0.893      0.931       0.76



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     75/300      1.14G     0.7496     0.5558      1.026         12        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.13it/s]

                   all       1720       3999      0.925      0.896      0.932       0.76



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     76/300      1.14G     0.7475     0.5431       1.03         14        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.935      0.886      0.931       0.76



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     77/300      1.14G      0.763     0.5573      1.037         13        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.924      0.894      0.931      0.762



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     78/300      1.14G     0.7528     0.5558      1.029         12        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.23it/s]

                   all       1720       3999      0.925      0.893      0.931      0.759



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     79/300      1.14G     0.7538     0.5493      1.031         11        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.923      0.898      0.933      0.764



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     80/300      1.14G     0.7405     0.5413      1.023         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.11it/s]

                   all       1720       3999      0.926      0.894      0.932      0.763



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     81/300      1.14G     0.7441     0.5492      1.025         16        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.937      0.882       0.93      0.762



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     82/300      1.14G     0.7517     0.5445      1.027         17        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.927       0.89       0.93      0.762



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     83/300      1.14G     0.7433     0.5441      1.025         18        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.927       0.89      0.931      0.763



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     84/300      1.14G      0.742     0.5455      1.023          8        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.03it/s]

                   all       1720       3999      0.935      0.885      0.931      0.765



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     85/300      1.14G     0.7334     0.5358      1.019          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.936      0.886      0.931      0.763



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     86/300      1.14G     0.7347     0.5405      1.018         11        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.928      0.891      0.931      0.764



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     87/300      1.14G     0.7329     0.5364       1.02         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.18it/s]

                   all       1720       3999      0.929      0.892      0.932      0.762



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     88/300      1.14G      0.744     0.5431      1.027         36        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.24it/s]

                   all       1720       3999      0.922      0.897      0.931      0.765



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     89/300      1.14G     0.7293     0.5313      1.014          4        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]

                   all       1720       3999      0.928      0.891      0.932      0.766



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     90/300      1.14G     0.7352     0.5386      1.023         14        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.933      0.887      0.931      0.765



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     91/300      1.14G     0.7318     0.5357      1.024         15        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.14it/s]

                   all       1720       3999      0.927      0.894      0.932      0.767



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     92/300      1.14G     0.7298      0.528      1.018         12        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.18it/s]

                   all       1720       3999      0.927      0.893      0.933      0.767



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     93/300      1.14G     0.7323     0.5267      1.016         12        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.22it/s]

                   all       1720       3999      0.925      0.893      0.932      0.766



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     94/300      1.14G     0.7348     0.5306       1.02         12        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.18it/s]

                   all       1720       3999      0.926      0.896      0.932      0.767



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     95/300      1.14G     0.7385     0.5341      1.023         12        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.22it/s]

                   all       1720       3999      0.933      0.889      0.932      0.766



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     96/300      1.14G     0.7222     0.5279      1.016          7        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.929      0.892      0.932      0.767



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     97/300      1.14G     0.7289     0.5297      1.019         12        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.22it/s]

                   all       1720       3999      0.927      0.894      0.932      0.767



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     98/300      1.14G     0.7233     0.5223      1.017          8        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.927      0.892      0.931      0.766



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


     99/300      1.14G     0.7352     0.5359      1.022          7        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.14it/s]

                   all       1720       3999      0.925      0.895      0.931      0.767



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    100/300      1.14G     0.7293     0.5302      1.016          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.13it/s]

                   all       1720       3999       0.93       0.89      0.931      0.768



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    101/300      1.14G     0.7134     0.5173      1.012         16        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.931      0.889      0.932      0.768



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    102/300      1.14G     0.7181     0.5226      1.011          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.931      0.893      0.932      0.769



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    103/300      1.14G     0.7242     0.5226      1.016         16        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.13it/s]

                   all       1720       3999      0.929       0.89      0.931      0.768



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    104/300      1.14G     0.7132      0.517      1.008         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]

                   all       1720       3999      0.929      0.894      0.933      0.767



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    105/300      1.14G     0.7223     0.5221      1.014         12        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.11it/s]

                   all       1720       3999      0.934      0.892      0.933      0.766



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    106/300      1.14G     0.7289     0.5226       1.02          9        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999       0.93      0.893      0.933      0.769



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    107/300      1.14G     0.7156     0.5121       1.01         11        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]

                   all       1720       3999      0.932      0.891      0.932      0.768



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    108/300      1.14G      0.718      0.519      1.013         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.931       0.89      0.932      0.769



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    109/300      1.14G     0.7161     0.5189      1.009         14        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.936      0.888      0.932       0.77



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    110/300      1.14G     0.7151     0.5169      1.011         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.22it/s]

                   all       1720       3999      0.934      0.889      0.931       0.77



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    111/300      1.14G     0.7206     0.5195      1.012          9        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.26it/s]

                   all       1720       3999      0.933       0.89      0.932       0.77



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    112/300      1.14G      0.713     0.5131      1.012          8        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.932       0.89      0.932      0.769



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    113/300      1.14G     0.7183     0.5218      1.013         14        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.931      0.892      0.933      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    114/300      1.14G     0.7018     0.5039      1.003         21        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.932      0.892      0.933      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    115/300      1.14G     0.7199      0.519      1.015         45        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:22<00:00,  9.44it/s]


                   all       1720       3999       0.93      0.894      0.934      0.773

      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    116/300      1.14G      0.704     0.5089      1.005         12        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.932      0.893      0.933      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    117/300      1.14G     0.7049     0.5071      1.007         11        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.933      0.891      0.933       0.77



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    118/300      1.14G     0.7191     0.5187      1.013          6        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.934       0.89      0.932      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    119/300      1.14G     0.6996     0.5036      1.007          7        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.935       0.89      0.933      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    120/300      1.14G     0.7086     0.5069      1.006          7        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.18it/s]

                   all       1720       3999      0.932       0.89      0.932      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    121/300      1.14G     0.7135     0.5081      1.005          3        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.933       0.89      0.933      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    122/300      1.14G     0.7042     0.5066      1.006         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.14it/s]

                   all       1720       3999      0.934      0.891      0.933      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    123/300      1.14G      0.703     0.5038          1          7        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.933       0.89      0.932      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    124/300      1.14G     0.7104     0.5051      1.009          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.932      0.891      0.932      0.771



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    125/300      1.14G      0.707     0.5102      1.006          7        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.21it/s]

                   all       1720       3999      0.934      0.888      0.933      0.773



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    126/300      1.14G     0.7109      0.513      1.007          8        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.23it/s]

                   all       1720       3999      0.933      0.889      0.933      0.774



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    127/300      1.14G     0.7079     0.5099      1.007          9        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.934       0.89      0.933      0.774



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    128/300      1.14G     0.6918     0.4943      1.001          8        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.22it/s]

                   all       1720       3999      0.931      0.893      0.933      0.773



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    129/300      1.14G     0.6998     0.5053      1.002         12        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.931      0.892      0.932      0.773



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    130/300      1.14G     0.6977     0.4945      1.002          8        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.34it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:20<00:00, 10.24it/s]

                   all       1720       3999      0.931      0.893      0.933      0.775



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    131/300      1.14G     0.6867     0.4944     0.9964         10        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.33it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.932      0.892      0.933      0.774



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    132/300      1.14G     0.7119       0.51      1.006          6        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.931      0.892      0.933      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    133/300      1.14G     0.6957     0.4985      1.005         13        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.931      0.893      0.933      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    134/300      1.14G     0.7047     0.5048      1.006          5        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.931      0.893      0.934      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    135/300      1.14G     0.7004     0.4975      1.008         15        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.14it/s]

                   all       1720       3999      0.934      0.891      0.934      0.772



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    136/300      1.14G     0.6991     0.4973      1.004         15        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.13it/s]

                   all       1720       3999      0.934      0.889      0.934      0.773



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    137/300      1.14G     0.6947     0.4922     0.9981         19        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.934       0.89      0.933      0.773



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    138/300      1.14G     0.7019     0.4944      1.002         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.934      0.891      0.934      0.774



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    139/300      1.14G     0.7032     0.5017     0.9992          6        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.11it/s]

                   all       1720       3999      0.932      0.893      0.933      0.774



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    140/300      1.14G     0.6738     0.4789     0.9893         10        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.929      0.893      0.933      0.774



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    141/300      1.14G      0.697     0.4939      1.001         14        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999      0.932      0.889      0.933      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    142/300      1.14G     0.6944     0.4928     0.9989          8        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.931       0.89      0.934      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    143/300      1.14G     0.6946     0.4917      1.002         12        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.19it/s]

                   all       1720       3999      0.928      0.894      0.934      0.777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    144/300      1.14G     0.6962     0.4942     0.9999          7        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999       0.93      0.892      0.934      0.777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    145/300      1.14G     0.7049     0.5062      1.003         10        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.931      0.892      0.934      0.778



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    146/300      1.14G     0.6942     0.4966     0.9991         16        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.932      0.893      0.934      0.777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    147/300      1.14G     0.6924     0.4903     0.9992         12        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.20it/s]

                   all       1720       3999      0.931      0.892      0.934      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    148/300      1.14G     0.6918     0.4902     0.9988          9        640: 100%|██████████| 1011/1011 [03:53<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.18it/s]

                   all       1720       3999      0.934       0.89      0.934      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    149/300      1.14G     0.6814     0.4862     0.9945          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.32it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.16it/s]

                   all       1720       3999      0.933      0.891      0.934      0.777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    150/300      1.14G     0.6893     0.4903     0.9927         15        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.01it/s]

                   all       1720       3999      0.933      0.891      0.934      0.777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    151/300      1.14G     0.6784     0.4824     0.9951         44        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.932       0.89      0.934      0.777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    152/300      1.14G     0.6786     0.4788     0.9907         20        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.17it/s]

                   all       1720       3999      0.932      0.891      0.934      0.777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    153/300      1.14G     0.6871     0.4834     0.9927         12        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.15it/s]

                   all       1720       3999       0.93      0.893      0.934      0.777



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    154/300      1.14G     0.6849     0.4848     0.9987         13        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.13it/s]

                   all       1720       3999      0.931      0.892      0.934      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    155/300      1.14G     0.6844     0.4824     0.9909         19        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.09it/s]

                   all       1720       3999      0.929      0.893      0.934      0.775



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    156/300      1.14G     0.6833     0.4848     0.9951         12        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.11it/s]

                   all       1720       3999      0.929      0.893      0.934      0.775



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    157/300      1.14G     0.6843     0.4876     0.9923         16        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]

                   all       1720       3999      0.928      0.894      0.934      0.775



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    158/300      1.14G     0.6846     0.4809     0.9966         11        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.30it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.09it/s]

                   all       1720       3999      0.931      0.892      0.934      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    159/300      1.14G     0.6825     0.4837     0.9935         13        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]

                   all       1720       3999      0.934      0.891      0.935      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    160/300      1.14G     0.6836     0.4884     0.9958          9        640: 100%|██████████| 1011/1011 [03:54<00:00,  4.31it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.12it/s]

                   all       1720       3999      0.933      0.891      0.935      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    161/300      1.14G     0.6776     0.4763     0.9922         14        640: 100%|██████████| 1011/1011 [03:55<00:00,  4.29it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:21<00:00, 10.14it/s]

                   all       1720       3999       0.93      0.893      0.935      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    162/300      1.14G     0.6873     0.4867     0.9983          7        640: 100%|██████████| 1011/1011 [04:12<00:00,  4.00it/s]
                 Class     Images  Instances      Box(P          R      mAP50  mAP50-95): 100%|██████████| 215/215 [00:34<00:00,  6.24it/s]

                   all       1720       3999      0.932      0.893      0.935      0.776



      Epoch    GPU_mem   box_loss   cls_loss   dfl_loss  Instances       Size


    163/300      1.14G     0.7289     0.4856      1.005         10        640:  17%|█▋        | 169/1011 [00:40<03:19,  4.22it/s]


KeyboardInterrupt: 

# 🧠 YOLOv8 Training Metrics Cheatsheet

## 🔢 Loss Curves

| Metric            | Meaning                                                | Goal / Good Trend                |
|-------------------|--------------------------------------------------------|----------------------------------|
| `train/box_loss`  | Bounding box regression loss (training)                | ↓ Steady decline                 |
| `train/cls_loss`  | Classification loss (training)                         | ↓ Should fall and stabilize     |
| `train/dfl_loss`  | Distribution Focal Loss (for bounding box quality)     | ↓ Smooth drop                   |
| `val/box_loss`    | Box loss on validation set                             | ↓ Best indicator of generalization |
| `val/cls_loss`    | Classification loss on validation                      | ↓ Falling → Good model fit      |
| `val/dfl_loss`    | DFL loss on validation                                 | ↓ Matches train trend = no overfit |

---

## 📈 Detection Metrics

| Metric                | Description                                                 | Goal / Interpretation           |
|------------------------|-------------------------------------------------------------|----------------------------------|
| `metrics/precision(B)` | % of correct detections out of all predictions              | ↑ High = few false positives     |
| `metrics/recall(B)`    | % of actual objects detected                                | ↑ High = few missed detections   |
| `metrics/mAP50(B)`     | Accuracy of boxes at 50% IoU threshold                     | ↑ Main object detection metric   |
| `metrics/mAP50-95(B)`  | Average mAP over IoU 0.5 to 0.95 (stricter)                | ↑ Robust performance measure     |

---

## 🧪 Ideal Curve Patterns

- **Losses**: should steadily decrease.
- **Validation loss** rising while training loss decreases → possible **overfitting**.
- **Precision & Recall**: should increase and plateau near 1.0.
- **mAP**: aim for **>0.90 mAP50** and **>0.80 mAP50-95** for high-quality results.

---

## ⚠️ Tips

- 🧊 Use `half=True` for mixed precision training on supported GPUs.
- 🧠 Watch `val/box_loss` and `mAP50-95` to catch overfitting early.
- 🧪 If metrics plateau early, try increasing `epochs`, `imgsz`, or improving labels.
- 💾 Check `runs/train/.../results.png` for curve visualization.

---
